checker/src/org/checkerframework/checker/initialization/InitializationVisitor.java

```diff
@@ -181,6 +184,19 @@ protected boolean checkContract(
                             return true;
                         }
                     }
+                } else if (fa.getReceiver() instanceof LocalVariable) {
+                    // TODO: what is the best way to handle other types of
+                    // receivers?
+                    Element elem = ((LocalVariable) fa.getReceiver()).getElement();
+                    AnnotatedTypeMirror recvType = atypeFactory.getAnnotatedType(elem);
```

Comment:
No, but you can extract the code in GenericAnnotatedTypeFactory#getAnnotationFromJavaExpressionString into such a method.